libyul/backends/wasm/EVMToEwasmTranslator.cpp

```diff
@@ -985,19 +1000,24 @@ function mload(x1, x2, x3, x4) -> z1, z2, z3, z4 {
 	z1, z2, z3, z4 := mload_internal(to_internal_i32ptr(x1, x2, x3, x4))
 }
 function mload_internal(pos:i32) -> z1, z2, z3, z4 {
-	z1 := endian_swap(i64.load(pos))
-	z2 := endian_swap(i64.load(i32.add(pos, 8:i32)))
-	z3 := endian_swap(i64.load(i32.add(pos, 16:i32)))
-	z4 := endian_swap(i64.load(i32.add(pos, 24:i32)))
+	z1 := bswap64(i64.load(pos))
+	z2 := bswap64(i64.load(i32.add(pos, 8:i32)))
+	z3 := bswap64(i64.load(i32.add(pos, 16:i32)))
+	z4 := bswap64(i64.load(i32.add(pos, 24:i32)))
+}
+function mload_address(pos:i32) -> z2, z3, z4 {
```

Comment:
We should start adding comments. Is this expecting to read 32 or 20 bytes from `pos`?

Reply:
This would read `24` bytes from `pos`. Yeah, we really need to add more comments in that file.